### ISSUE TITLE
[Feature] Background image

### ISSUE THREAD

**Issue by brantje** [PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor#39]
Hi!
Thanks for your component, love it!
I have a small feature request. 
Would it be possible to add an background image to the map?
Reason for this is i have drawn my house layout instead of the map, and would love to see the position of the vacuum real time.

**Comment by PiotrMachowski** [comment]
>I have a small feature request.
It's definitely not small 😉

**Comment by brantje** [comment]
I'll try to give it a go later today.

**Comment by PiotrMachowski** [comment]
I plan to add this feature to the [map card](https://github.com/PiotrMachowski/lovelace-xiaomi-vacuum-map-card), is it enough?

**Comment by brantje** [comment]
Yes, definitely. As long the background image matches the map size, i guess it won't be a problem.

**Comment by PiotrMachowski** [comment]
It's not possible to be sure that sizes are matching, which makes situation harder. It will require calibration similar like one already existing in a [map card](https://github.com/PiotrMachowski/lovelace-xiaomi-vacuum-map-card)

**Comment by DirkFust** [comment]
I second this request. I'd love to put a cleaned image of my flat below the camera from your Xiaomi Cloud Map Extractor and have it trace the path on it.
Perhaps if you skip the calibration at first we could fiddle with the image to line it up. It won't be perfect, but I'd love to have "good enough" ;)

Thanks for your awesome work!

**Comment by julianrinaldi** [comment]
I third this request. It would make my setup amazing. Sent you a tip! Probably my new favorite integration.

**Comment by Graeme-SStrong** [comment]
I've achieved this with a work around by putting this into a picture elements card and using mix-blend-mode. 

`            - type: image
              entity: camera.xiaomi_cloud_map_extractor
              camera_image: "${states['vacuum.roborock_s5_max'].state === 'docked' ? '' : 'camera.xiaomi_cloud_map_extractor'}"
              style: 
                transform: rotate(180deg) scale(1.25)
                mix-blend-mode: lighten
                top: -23%
                left: -2.55%
                opacity: 80%
              action: none `

In configuration.yaml, I set all colors to [0,0,0,0], except for the walls and the elements I want to see (no go zones, path, robo, etc), which I set to a bright colors. Once everything is lined up, set walls color to [0,0,0,0] to hide them.

This hides the map when the vacuum is docked and displays its path and zones when it's active.

**Comment by RedFoxy** [comment]
Cool! I hope to see it soon!

**Comment by tikel1** [comment]
+1
any news on that?

**Comment by mihatsu** [comment]
I was able to do this with some minor modifications. I'll try polish it slightly then release how 
https://github.com/humbertogontijo/homeassistant-roborock/issues/388#issuecomment-1608519546
